File: src/lib/api/taskbar.ts
import { invoke } from '@tauri-apps/api/core';
import type {
  TaskbarPreferenceMode,
  TaskbarSettings,
  TaskbarSnapshot,
  TaskbarStatus,
} from '$lib/types';

export async function getTaskbarSettings(): Promise<TaskbarSettings> {
  return invoke('get_taskbar_settings');
}

export async function getTaskbarStatus(): Promise<TaskbarStatus> {
  return invoke('get_taskbar_status');
}

export async function setTaskbarPlayerEnabled(enabled: boolean): Promise<TaskbarStatus> {
  return invoke('set_taskbar_player_enabled', { enabled });
}

export async function setTaskbarPlayerMode(mode: TaskbarPreferenceMode): Promise<TaskbarStatus> {
  return invoke('set_taskbar_player_mode', { mode });
}

export async function setTaskbarPlayerOffset(offsetX: number): Promise<TaskbarStatus> {
  return invoke('set_taskbar_player_offset', { offsetX });
}

export async function setTaskbarPlayerDisplayOptions(
  showTitleMarquee: boolean,
  showProgress: boolean,
): Promise<TaskbarSettings> {
  return invoke('set_taskbar_player_display_options', { showTitleMarquee, showProgress });
}

export async function setTaskbarPlayerMiniModeBehavior(
  hideInMiniPlayer: boolean,
): Promise<TaskbarSettings> {
  return invoke('set_taskbar_player_mini_mode_behavior', { hideInMiniPlayer });
}

export async function setTaskbarPlayerVisible(visible: boolean): Promise<TaskbarStatus> {
  return invoke('set_taskbar_player_visible', { visible });
}

export async function updateTaskbarPlayer(snapshot: TaskbarSnapshot): Promise<void> {
  return invoke('update_taskbar_player', { snapshot });
}

File: src-tauri/src/commands/taskbar.rs
use musicplayer_taskbar_helper::{TaskbarPreferenceMode, TaskbarSnapshot};
use tauri::{AppHandle, State};

use crate::error::AppError;
use crate::taskbar::{TaskbarManager, TaskbarSettings, TaskbarStatus};

#[tauri::command]
pub fn get_taskbar_settings(manager: State<TaskbarManager>) -> Result<TaskbarSettings, AppError> {
    manager.settings()
}

#[tauri::command]
pub fn get_taskbar_status(manager: State<TaskbarManager>) -> Result<TaskbarStatus, AppError> {
    manager.status()
}

#[tauri::command]
pub fn set_taskbar_player_enabled(
    enabled: bool,
    app: AppHandle,
    manager: State<TaskbarManager>,
) -> Result<TaskbarStatus, AppError> {
    manager.set_enabled(&app, enabled)
}

#[tauri::command]
pub fn set_taskbar_player_mode(
    mode: TaskbarPreferenceMode,
    app: AppHandle,
    manager: State<TaskbarManager>,
) -> Result<TaskbarStatus, AppError> {
    manager.set_mode(&app, mode)
}

#[tauri::command]
pub fn set_taskbar_player_offset(
    offset_x: i32,
    app: AppHandle,
    manager: State<TaskbarManager>,
) -> Result<TaskbarStatus, AppError> {
    manager.set_offset_x(&app, offset_x)
}

#[tauri::command]
pub fn set_taskbar_player_display_options(
    show_title_marquee: bool,
    show_progress: bool,
    manager: State<TaskbarManager>,
) -> Result<TaskbarSettings, AppError> {
    manager.set_display_options(show_title_marquee, show_progress)
}

#[tauri::command]
pub fn set_taskbar_player_mini_mode_behavior(
    hide_in_mini_player: bool,
    manager: State<TaskbarManager>,
) -> Result<TaskbarSettings, AppError> {
    manager.set_mini_player_behavior(hide_in_mini_player)
}

#[tauri::command]
pub fn set_taskbar_player_visible(
    visible: bool,
    app: AppHandle,
    manager: State<TaskbarManager>,
) -> Result<TaskbarStatus, AppError> {
    manager.set_visible(&app, visible)
}

#[tauri::command]
pub fn update_taskbar_player(
    snapshot: TaskbarSnapshot,
    app: AppHandle,
    manager: State<TaskbarManager>,
) -> Result<(), AppError> {
    manager.update(&app, snapshot)
}
